### DISCUSSION TITLE
Yarn 2 not being ignored by default

### DISCUSSION THREAD

**Discussion by yordis** [github-linguist/linguist#5738] [Q&A] [answered]
Hey there,

I am a bit confused about my project,

I recently had to https://github.com/straw-hat-team/javascript-monorepo/blob/7569a753a785acb272d4bb0122085862a1f6e176/.gitattributes#L1-L3 which doesn't seem to work perfectly (or it doesn't at all) with Yarn 2 and the configuration I set up.

Notice that the count of JavaScript files,

![Screen Shot 2022-01-19 at 6 07 24 PM](https://user-images.githubusercontent.com/4237280/150232824-0ddb39c2-ae8b-4335-a503-681fe2a3383d.png)

Once I click on the JavaScript one, a lot of files from inside `.yarn` are not being ignored.

I expected the following to work:

https://github.com/github/linguist/blob/80f3531e8a1014a23f4606458e5a528053ed3cac/lib/linguist/vendor.yml#L44-L49

https://github.com/github/linguist/blob/80f3531e8a1014a23f4606458e5a528053ed3cac/lib/linguist/generated.rb#L415-L417

Is this particular package being used at GitHub?

If so, is it released with such changes or how can I verify that it works since I needed to manually do these things in my project?

Thanks in advance,

**Comment by lildude** [comment, marked as answer]
> Once I click on the JavaScript one, a lot of files from inside `.yarn` are not being ignored.

This is expected. As I said in the issue in which your first asked:

> This override should do the trick to mark everything as generated, but it won't affect the Search results returned when you click the languages in the sidebar. It only affects the percentages.

When you click the language you enter the search results for that language in your repo (note the URL).

Linguist overrides deliberately don't affect Search as you could end up never being able to find any of the files in your repo.

> Is this particular package being used at GitHub?

It is.

> If so, is it released with such changes or how can I verify that it works since I needed to manually do these things in my project?

It is. See the README.md for details on how you can install it and run it locally.
